Agents/HMLandRegistryAgent/agent/kgutils/stackclients.py

```diff
@@ -0,0 +1,288 @@
+        
+        Initially, "name", "iri", and "endpoint" were required for the FeatureInfoAgent
+        to work (i.e. be able to retrieve data from PostGIS). Strictly, those are no 
+        longer needed, but kept for consistency with previous implementations
+        Further properties can be added as needed, i.e. for styling purposes
+
+        Arguments:
+            building_iri {str} -- IRI of building/property to be uploaded to PostGIS
+            value_estimate {str} -- Market value estimate of building (in GBP)
+        """
+        
+        if building_iri:
+            # Condition data to upload (incl. columns (initially?) required by DTVF)
+            props = {
+                'iri': building_iri,
+                'endpoint': kg_endpoint,
+                'name': None,
+                'price': None
+            }
+            try:
+                props['name'] = building_iri.split('/')[-1].replace('>','')
+            except Exception:
+                pass
+            # Property value estimate (upload required for styling purposes)
+            try:
+                props['price'] = float(value_estimate)
+            except Exception:
+                pass
+            
+            # Prepare SQL update string (remove keys with empty values first)
+            props = {k: v for k,v in props.items() if v is not None}
+            cols = f"( {', '.join(props.keys())} )"
+            vals = "VALUES ("
+            for v in props.values():
+                if isinstance(v, str):
+                    vals += f"'{v}', "
+                elif isinstance(v, float):
+                    vals += f"{v}, "
+            vals = vals[:-2]
+            vals += ")"
+
+            # Create SQL update
+            try:
+                with jaydebeapi.connect(*self.conn_props) as conn:
+                    # Create a cursor object ...
+                    with conn.cursor() as curs:
+                        # ... and execute the SQL query
+                        # When DO UPDATE is specified, a special virtual table called EXCLUDED
+                        # is available for use within the UPDATE clause. The table contains 
+                        # the values suggested in the original INSERT command (i.e. the ones 
+                        # conflicting with the existing table values)
+                        sql_update = f"INSERT INTO \"{table}\" {cols} {vals} \
+                                       ON CONFLICT (iri) DO UPDATE SET \
+                                        price = EXCLUDED.price"
+                        curs.execute(sql_update)
+                        if curs.rowcount != 1:
+                            raise StackException(f'Expected to update 1 row; however {curs.rowcount} rows affected.')
+            except Exception as ex:
+                logger.error(f'Unsuccessful JDBC interaction: {ex}')
+                raise StackException('Unsuccessful JDBC interaction.') from ex
+
+
+class GeoserverClient(StackClient):
```

Comment:
@gpeb2:  Could you please have a quick cross-check of the `stackclients.py`, especially the creation of the virtual table using GeoServer. Does this look alright to you?

I have tested the agent with a sample set of ~500 buildings and it seems to work as expected (i.e. layers get created in PostGIS and Geoserver, PostGIS gets populated, Geoserver layer shows building footprints in `OpenLayers3` preview). The only issue: I did not get the WMS endpoint visualised using the DTVF (all other WMS endpoints served by Geoserver work though). 
Do I have to specify any styling or so? Otherwise, this might "just" be a Geoserver issue; however, I was not able to restart Geoserver (as the Docker registry seems down again)